### PR TITLE
getAll getById for admin and user

### PR DESCRIPTION
**Pull request by OlegKandalov**
-added service RoleCheckService
- added logic where user cannot take deleted elements

### CODE REVIEW

File: src/main/java/com/exadel/discount/platform/web/CategoryController.java [hunk 1]
```diff
@@ -18,16 +19,26 @@
...
+    private final RoleCheckService roleCheckService;
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
     @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
-    public ResponseEntity<List<CategoryResponseDto>> getAllCategories() {
-        return new ResponseEntity<>(categoryService.getAll(), HttpStatus.OK);
+    public ResponseEntity<List<CategoryResponseDto>> getAllCategories(
+            @RequestParam(value = "isDeleted", required = false, defaultValue = "false") boolean isDeleted
+    ) {
+        if (isDeleted & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/CategoryController.java:29]: `&&`

File: src/main/java/com/exadel/discount/platform/web/CategoryController.java [hunk 2]
```diff
@@ -18,16 +19,26 @@
...
+        }
+        return new ResponseEntity<>(categoryService.getAll(isDeleted), HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
-    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
+    @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
     public ResponseEntity<CategoryResponseDto> getCategoryById(@PathVariable UUID id) {
+        CategoryResponseDto categoryDto = categoryService.getById(id);
+        if (categoryDto.isDeleted() & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/CategoryController.java:39]: `&&`

File: src/main/java/com/exadel/discount/platform/web/SubCategoryController.java [hunk 3]
```diff
@@ -23,16 +24,26 @@
...
+    private final RoleCheckService roleCheckService;
 
     @GetMapping(produces = APPLICATION_JSON_VALUE)
     @PreAuthorize("hasAnyAuthority('ADMINISTRATOR','USER')")
-    public ResponseEntity<List<SubCategoryResponseDto>> getAllSubCategories() {
-        return new ResponseEntity<>(subCategoryService.getAll(), HttpStatus.OK);
+    public ResponseEntity<List<SubCategoryResponseDto>> getAllSubCategories(
+            @RequestParam(value = "isDeleted", required = false, defaultValue = "false") boolean isDeleted
+    ) {
+        if (isDeleted & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/SubCategoryController.java:34]: `&&`

File: src/main/java/com/exadel/discount/platform/web/SubCategoryController.java [hunk 4]
```diff
@@ -23,16 +24,26 @@
...
+        }
+        return new ResponseEntity<>(subCategoryService.getAll(isDeleted), HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}", produces = APPLICATION_JSON_VALUE)
-    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
+    @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
     public ResponseEntity<SubCategoryResponseDto> getSubCategoryById(@PathVariable UUID id) {
+        SubCategoryResponseDto subCategoryDto = subCategoryService.getById(id);
+        if (subCategoryDto.isDeleted() & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/SubCategoryController.java:44]: `&&`

File: src/main/java/com/exadel/discount/platform/web/VendorController.java [hunk 5]
```diff
@@ -18,16 +19,26 @@
...
+    private final RoleCheckService roleCheckService;
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
     @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
-    public ResponseEntity<List<VendorResponseDto>> getAllVendors() {
-        return new ResponseEntity<>(vendorService.getAll(), HttpStatus.OK);
+    public ResponseEntity<List<VendorResponseDto>> getAllVendors(
+            @RequestParam(value = "isDeleted", required = false, defaultValue = "false") boolean isDeleted
+    ) {
+        if (isDeleted & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/VendorController.java:29]: `&&`

File: src/main/java/com/exadel/discount/platform/web/VendorController.java [hunk 6]
```diff
@@ -18,16 +19,26 @@
...
+        }
+        return new ResponseEntity<>(vendorService.getAll(isDeleted), HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
-    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
+    @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
     public ResponseEntity<VendorResponseDto> getVendorById(@PathVariable UUID id) {
+        VendorResponseDto vendorResponseDto = vendorService.getById(id);
+        if (vendorResponseDto.isDeleted() & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/VendorController.java:39]: `&&`

File: src/main/java/com/exadel/discount/platform/web/VendorLocationController.java [hunk 7]
```diff
@@ -19,16 +20,26 @@
...
+    private final RoleCheckService roleCheckService;
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
     @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
-    public ResponseEntity<List<VendorLocationResponseDto>> getAllLocations() {
-        return new ResponseEntity<>(locationService.getAll(), HttpStatus.OK);
+    public ResponseEntity<List<VendorLocationResponseDto>> getAllLocations(
+            @RequestParam(value = "isDeleted", required = false, defaultValue = "false") boolean isDeleted
+    ) {
+        if (isDeleted & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/VendorLocationController.java:30]: `&&`

File: src/main/java/com/exadel/discount/platform/web/VendorLocationController.java [hunk 8]
```diff
@@ -19,16 +20,26 @@
...
+        }
+        return new ResponseEntity<>(locationService.getAll(isDeleted), HttpStatus.OK);
     }
 
     @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
-    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
+    @PreAuthorize("hasAnyAuthority('ADMINISTRATOR', 'USER')")
     public ResponseEntity<VendorLocationResponseDto> getLocationById(@PathVariable UUID id) {
+        VendorLocationResponseDto locationResponseDto = locationService.getById(id);
+        if (locationResponseDto.isDeleted() & roleCheckService.isUser()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/web/VendorLocationController.java:40]: `&&`

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

**Review by ohalushchak-exadel — APPROVED** [review]

**Review by nkharitoniuk — APPROVED** [review]

**Pull request merged**